fred2/briefingeditordlg.cpp

```diff
@@ -1091,23 +1093,30 @@ void briefing_editor_dlg::OnMakeIcon()
             biconp->type = ICON_ASTEROID_FIELD;
 	}
 	// jumpnodes
-	else if(jnp != Jump_nodes.end()){
-		// find the first navbuoy
+	else if (jump_node_found) {
 		biconp->ship_class = -1;
+
+		// Kept for reference, the following block causes crashes, and was also a likely copy paste error from navbouys. 
+		// The best I can tell, they forgot to change this section when they changed to biconp from iconp -- Cyborg17
```

Comment:
You still haven't told me what the crash is.  Is it a C++ exception?  Is it a memory access violation?  What's the error message?  What line does it occur on?

That "the value may be negative" line just means that negative distances are allowed, such as distance from the *end* of a vector as well as from the beginning.  Since the arguments are in the correct order here, the distance should not be negative.

If `cbegin` and `cend` are the problem, then you could try substituting `begin` and `end` to see if that makes a difference.  And if `iconp` is not used for anything, then you could try deleting the variable declaration and see if that causes any problems.

The reason to iterate over ship info for a jump node is obvious.  The briefing pop-up display is going to show a navbuoy, so the code is looking for the first navbuoy listed in ships.tbl.

Reply:
There are plenty of *working* examples of using `std::distance(some_vector.[c]begin(), it)` throughout the codebase, so obviously the mere fact that it's being done here isn't the real problem.

Reply:
> The reason to iterate over ship info for a jump node is obvious. The briefing pop-up display is going to show a navbuoy, so the code is looking for the first navbuoy listed in ships.tbl.

Ah!  I'm sorry!  I completely forgot about this functionality.  Now everything makes sense.  I really thought someone had just hastily changed this in the past and no one caught the mistake, since it's Fred and not FSO.  

The crash is a ctd without any message.  Since I'm wrong about the right side of that line, my current theory is that `iconp->ship_class` is just not initialized correctly or `iconp` is a `nullptr` in this code path.  (The same crash occurs for both way points and jump nodes, by the way)

The other function which uses iconp for the Change_locally functionality seems to work fine, even when working with jump node or waypoint icons.  So I'm leaning towards `iconp->ship_class' not being initialized correctly.  (Again, this is the only other place this pointer is used, so I do not believe it's a risk to leave it as it is in that other function)

All right, I just made changes (which I have tested) which perform the desired functionality.
So now, in addition to the initial message, this PR will also guarantee that Jump Nodes and Waypoints will show up as navbuoys by default when clicked on in briefings.

Reply:
Should be done as you requested.